rts/Lua/LuaMathExtra.cpp

```diff
@@ -184,6 +185,39 @@ int LuaMathExtra::smoothstep(lua_State* L) {
 	return 1;
 }
 
+
+/*** Returns the normalize vector of an given vector.
+ *
+ * @function math.normalize
+ * @number x
+ * @number[opt] xn and so on
+ * @treturn zero xn numbers|xn numbers
+ */
+int LuaMathExtra::normalizevector(lua_State* L)
+{
+	RECOIL_DETAILED_TRACY_ZONE;
+	lua_Number res = 0.0f;
+	const int param = lua_gettop(L);
+	for (int i = param; i >= 1; i--) {
+		res += Square(luaL_checknumber_noassert(L, i));
+	}
+	if likely(res > float3::nrm_eps())
+		res = math::sqrt(res);
+	if (res == 0)
+	{
+		for (int i = param; i >= 1; i--) {
+			lua_pushnumber(L, 0);
+		}
+		return param;
+	}
+	for (int i = param; i >= 1; i--) {
```

Comment:
Currently pushes backwards
```suggestion
	for (int i = 1; <= param; ++i) {
```